### PR TITLE
Fix exception when redirect url only differs by case

### PR DESCRIPTION
**Pull request by yufeih**

### CODE REVIEW

File: src/docfx/build/redirection/RedirectionMap.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public static (List<Error> errors, RedirectionMap map) Create(Docset docset)
             AddRedirections(docset.Config.Redirections);
 
             var redirectionsByRedirectionUrl = redirections
-                .GroupBy(file => file.RedirectionUrl)
+                .GroupBy(file => file.RedirectionUrl, PathUtility.PathComparer)
```
Comment (OsmondJiang) [inline src/docfx/build/redirection/RedirectionMap.cs]: >RedirectionUrl [](start = 38, length = 14)

the redirection url should be always case insenstive?

Comment (yufeih) [reply src/docfx/build/redirection/RedirectionMap.cs]: URL is also using os case sensitivity. We are a static site generator, that means a URL maps directly to a file, including case sensitivity.

Comment (OsmondJiang) [reply src/docfx/build/redirection/RedirectionMap.cs]: what if the url is absolute path?

---
In reply to: [201203096](https://github.com/dotnet/docfx/pull/3027#discussion_r201203096) [](ancestors = 201203096)

Comment (OsmondJiang) [reply src/docfx/build/redirection/RedirectionMap.cs]: and most of the browser will lower case url

---
In reply to: [201205938](https://github.com/dotnet/docfx/pull/3027#discussion_r201205938) [](ancestors = 201205938,201203096)

Comment (yufeih) [reply src/docfx/build/redirection/RedirectionMap.cs]: browser will not adjust URL casing.

Comment (OsmondJiang) [reply src/docfx/build/redirection/RedirectionMap.cs]: the host name will be lower case

---
In reply to: [201206103](https://github.com/dotnet/docfx/pull/3027#discussion_r201206103) [](ancestors = 201206103)

**Pull request merged**